File: src/php/Dhrystone/DhrystoneBenchmark.php
<?php

namespace EntityFX\NetBenchmark\Core\Dhrystone {

    use EntityFX\NetBenchmark\Core\Writer;
    use EntityFX\NetBenchmark\Core\Generic\BenchmarkBase;

    class DhrystoneBenchmark extends BenchmarkBase {

        private $dhrystone;

        public function __construct($writer, $printToConsole)
        {
            parent::__construct($writer, $printToConsole);
            $this->Ratio = 50;
            $this->dhrystone = new Dhrystone2($printToConsole);
        }

        public function Warmup($aspect = 0.05) {
        }

        protected function BenchImplementation() {
            return $this->dhrystone->bench();
        }

        protected function PopulateResult($benchResult, $result) {
            $benchResult["Points"] = $result["VaxMips"] * $this->Ratio;
            $benchResult["Result"] = $result["VaxMips"];
            $benchResult["Units"] = "DMIPS";
            $benchResult["Output"] = $result["Output"];
            return $benchResult;
        }
    }
}

File: src/lua/dhrystone/dhrystoneBenchmark.lua
require "dhrystone/dhrystone"

DhrystoneBenchmark = class(BenchmarkBase, function(a, writer, printToConsole)
    BenchmarkBase.init(a, writer, printToConsole)
    a.name = "DhrystoneBenchmark"
    a.ratio = 4
    a.dhrystone = Dhrystone2(false)
end)

function DhrystoneBenchmark:benchImplementation()
    return self.dhrystone:bench()
end

function DhrystoneBenchmark:warmup(aspect)

end

function DhrystoneBenchmark:populateResult(benchResult, result)
    -- dumpTable(result, 200)
    benchResult.points = result.VaxMips * self.ratio
    benchResult.result = result.VaxMips
    benchResult.units = "DMIPS"
    benchResult.output = result.Output
    return benchResult
end